modules/project/service-agents.tf

```diff
@@ -16,6 +16,17 @@
 
 # tfdoc:file:description Service agents supporting resources.
 
+# debug block
+
+# resource "terraform_data" "precondition" {
+#   lifecycle {
+#     precondition {
+#       condition     = local.primary_service_agents == null
+#       error_message = yamlencode(local.primary_service_agents)
+#     }
+#   }
+# }
+
```

Comment:
Do we want to keep this?

Reply:
Yes, until we refactor these locals which are **very** opaque. I explicitly mention adding this in the PR description. :)